repo2docker/buildpacks/base.py

```diff
@@ -125,7 +125,14 @@
 # of the repository but don't access any files in the repository. By executing
 # them before copying the repository itself we can cache these steps. For
 # example installing APT packages.
-{% for sd in pre_assemble_script_directives -%}
+{% if preassemble_script_files -%}
+# If scripts required during build are present, copy them
+{% for src, dst in preassemble_script_files.items() %}
```

Comment:
use `sorted(preassemble_script_files.items())` to ensure that files are added in a stable order. Dictionary randomization could potentially result in cache busting here for the same collection of files.

Reply:
Good catch! I added two more `sorted()` statements to the template for other loops over `items()`.